tests/integration/cli/src/link_code.rs

```diff
@@ -0,0 +1,70 @@
+use crate::assets::*;
+use anyhow::bail;
+use std::path::PathBuf;
+use std::process::Command;
+
+/// Data used to run a linking command for generated artifacts.
+#[derive(Debug)]
+pub struct LinkCode {
+    /// The directory to operate in.
+    pub current_dir: PathBuf,
+    /// Path to the linker used to run the linking command.
+    pub linker_path: PathBuf,
+    /// String used as an optimization flag.
+    pub optimization_flag: String,
+    /// Paths of objects to link.
+    pub object_paths: Vec<PathBuf>,
+    /// Path to the output target.
+    pub output_path: PathBuf,
+    /// Path to the static libwasmer library.
+    pub libwasmer_path: PathBuf,
+}
+
+impl Default for LinkCode {
+    fn default() -> Self {
+        #[cfg(not(windows))]
+        let linker = "cc";
+        #[cfg(windows)]
+        let linker = "clang";
+        Self {
+            current_dir: std::env::current_dir().unwrap(),
+            linker_path: PathBuf::from(linker),
+            optimization_flag: String::from("-O2"),
+            object_paths: vec![],
+            output_path: PathBuf::from("a.out"),
+            libwasmer_path: get_libwasmer_path(),
+        }
+    }
+}
+
+impl LinkCode {
+    pub fn run(&self) -> anyhow::Result<()> {
+        let mut command = Command::new(&self.linker_path);
+        let command = command
+            .current_dir(&self.current_dir)
+            .arg(&self.optimization_flag)
+            .args(
+                self.object_paths
+                    .iter()
+                    .map(|path| path.canonicalize().unwrap()),
+            )
+            .arg(&self.libwasmer_path.canonicalize()?);
+        #[cfg(windows)]
+        let command = command.arg("-luserenv").arg("-lWs2_32").arg("-ladvapi32");
```

Comment:
It seems this logic is duplicated?

Reply:
Yeah, this is test code.  I'm not sure where a good place to share this code would be.

We could maybe make it public in the `wasmer-cli` crate and then depend on it, but I think it's probably fine to duplicate

Those flags are required to link with `wasmer compile`, but since we do that in create-exe we also do that